Migrate to App Extensions API.

dbart01 opened the pull request:
#### What this does
- Switches from using `channelID` to `appID`
- Adjusts API endpoints for products and collections to use the new `/api/apps/...` endpoints
- Adjusts mocked response keys for new endpoints
- Adjusts the `BUYClient` initializer to accept an `appID`
- Adjusts tests for new endpoints

Fixes #124 

@bgulanowski @davidmuzi

bgulanowski commented:
Tests are failing. You will have to download the logs and have a look to see which tests are failing.

davidmuzi commented on Mobile Buy SDK/Mobile Buy SDK/Models/BUYCheckout.h:
```diff
@@ -225,7 +220,7 @@
- *  The unique identifier for the source: the channelId
+ *  The unique identifier for the source: the appId
  */
 @property (nonatomic, copy, readonly) NSString *sourceIdentifier;
```
@maximebedard when using app extensions, is using the `app ID` as the `source_identifier` correct?

maximebedard replied:
It's no longer necessary to send when you create/update checkouts, it will be inferred from the api client making the request.

dbart01 replied:
Is that true for `source_name` as well? @maximebedard

maximebedard replied:
What were you sending for source_name, the idea is to remove this field in the long run.

bgulanowski replied:
Seems to currently be `mobile_app`.

maximebedard replied:
Ok, leave it to mobile_app for now, but I believe this at one point will no longer be necessary. No ETA on this though.

dbart01 commented:
Added `appId` parameter in BuildKite to support new API endpoints. Removed one integration test that relied on `channel_id` for testing since we longer have a `channel_id`.

bgulanowski commented on Mobile Buy SDK/Mobile Buy SDK Tests/BUYClientTestBase.m:
```diff
@@ -29,7 +29,7 @@
 NSString * const BUYShopDomain_Placeholder = @"test_shop";
 NSString * const BUYAPIKey_Placeholder = @"channel_id";
-NSString * const BUYChannelId_Placeholder = @"api_key";
+NSString * const BUYAppId_Placeholder = @"api_key";
```
We should fix this naming problem. `AppId`->"api_key" and `APIKiey` -> "channel_id".

dbart01 replied:
Fixed.

bgulanowski commented:
Looks good. :shipit:
